src/schedule/dma_single_chan_domain.c

```diff
@@ -370,27 +371,29 @@ static void dma_single_chan_domain_unregister(struct ll_schedule_domain *domain,
 
 	/* check if task should be unregistered */
 	if (!pipe_task->registrable)
-		return;
+		return -EINVAL;
 
 	/* channel not registered */
 	if (!data->channel)
-		return;
+		return -EINVAL;
 
 	/* unregister domain owner */
 	if (dma_domain->owner == core) {
 		dma_domain_unregister_owner(domain, data);
-		return;
+		return 0;
 	}
 
 	/* some channel still running, so return */
 	if (dma_chan_is_any_running(dmas, dma_domain->num_dma))
-		return;
+		return -EINVAL;
```

Comment:
There are a lot more error paths now. This likely means more cause for failure and would be good to see this when debugging in the logs.

Reply:
> There are a lot more error paths now. This likely means more cause for failure and would be good to see this when debugging in the logs.

I just use the returned value to check if the unregistering is success or not for the subsequent task/client status update, so only using the same '-EINVAL' to denote this at the moment. We can add more debugging logs later if we do need to debug this part.

Reply:
Makes sense, let me change.